src/coreclr/vm/corelib.h

```diff
@@ -259,11 +259,36 @@ DEFINE_FIELD(DELEGATE,            METHOD_PTR_AUX,         _methodPtrAux)
 DEFINE_METHOD(DELEGATE,             CONSTRUCT_DELEGATE,     DelegateConstruct,          IM_Obj_IntPtr_RetVoid)
 DEFINE_METHOD(DELEGATE,             GET_INVOKE_METHOD,      GetInvokeMethod,            IM_RetIntPtr)
 
-DEFINE_CLASS(INT128,               System,                 Int128)
-DEFINE_CLASS(UINT128,              System,                 UInt128)
+DEFINE_CLASS(DOUBLE_SPC,            System,                 Double)
+DEFINE_METHOD(DOUBLE_SPC,           CONVERT_TO_INT32,       ConvertToInt32,             NoSig)
+DEFINE_METHOD(DOUBLE_SPC,           CONVERT_TO_UINT64,      ConvertToUInt64,            NoSig)
```

Comment:
```suggestion
// Defined as a primitive type, avoid redefinition
// DEFINE_CLASS(DOUBLE,            System,                 Double)
DEFINE_METHOD(DOUBLE,           CONVERT_TO_INT32,       ConvertToInt32,             NoSig)
DEFINE_METHOD(DOUBLE,           CONVERT_TO_UINT64,      ConvertToUInt64,            NoSig)
...
```
Would this work?

Reply:
Doing this fails to build:
```cs
H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\CreateRuntimeRootILLinkDescriptorFile.targets(41,5): error MSB4018: Nastąpiło nieoczekiwane niepowodzenie zadania „CreateRuntimeRootILLinkDescriptorFile”. [H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\CreateRuntimeRootILLinkDescriptorFile.targets(41,5): error MSB4018: System.Collections.Generic.KeyNotFoundException->Microsoft.Build.Framework.BuildException.GenericBuildTransferredException: The given key 'DOUBLE' was not present in the dictionary. [H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\CreateRuntimeRootILLinkDescriptorFile.targets(41,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.get_Item(TKey key) [H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\CreateRuntimeRootILLinkDescriptorFile.targets(41,5): error MSB4018:    at ILLink.Tasks.CreateRuntimeRootILLinkDescriptorFile.AddMethod(String methodName, String classId, String classNamespace, String className, Nullable`1 featureSwitch) in H:\Projects\dotnet\runtime\src\tools\illink\src\ILLink.Tasks\CreateRuntimeRootDescriptorFile.cs:line 438 [H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\CreateRuntimeRootILLinkDescriptorFile.targets(41,5): error MSB4018:    at ILLink.Tasks.CreateRuntimeRootILLinkDescriptorFile.ProcessMscorlib(String typeFile) in H:\Projects\dotnet\runtime\src\tools\illink\src\ILLink.Tasks\CreateRuntimeRootDescriptorFile.cs:line 241 [H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\CreateRuntimeRootILLinkDescriptorFile.targets(41,5): error MSB4018:    at ILLink.Tasks.CreateRuntimeRootILLinkDescriptorFile.Execute() in H:\Projects\dotnet\runtime\src\tools\illink\src\ILLink.Tasks\CreateRuntimeRootDescriptorFile.cs:line 136 [H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\CreateRuntimeRootILLinkDescriptorFile.targets(41,5): error MSB4018:    at Microsoft.Build.CommandLine.OutOfProcTaskAppDomainWrapperBase.InstantiateAndExecuteTask(IBuildEngine oopTaskHostNode, LoadedType taskType, String taskName, String taskLocation, String taskFile, Int32 taskLine, Int32 taskColumn, IDictionary`2 taskParams) [H:\Projects\dotnet\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

Reply:
The following should fix this build break:
```
#if FOR_ILLINK
DEFINE_CLASS(DOUBLE,            System,                 Double)
#endif
DEFINE_METHOD(DOUBLE,           CONVERT_TO_INT32,       ConvertToInt32,             NoSig)
...
```